src/main/java/org/semver4j/internal/range/processor/XRangeProcessor.java

```diff
@@ -19,10 +21,12 @@
  * <br>
  */
 public class XRangeProcessor implements Processor {
+    @NotNull
```

Comment:
That's not necessary in favor of field object assignment.

Reply:
Reasons 1 & 6